src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp

```diff
@@ -797,7 +797,7 @@ namespace {
                         const auto& oil_rate = properties->OilRate;
                         const auto& water_rate = properties->WaterRate;
                         const auto& gas_rate = properties->GasRate;
-                        if ((oil_rate.get<double>() + water_rate.get<double>() + gas_rate.get<double>()) == 0) {
+                        if (oil_rate.zero() && water_rate.zero() && gas_rate.zero()) {
```

Comment:
It is not related to this PR directly, while maybe later we should provide a better solution here. Just leaving a note here for discussion.  

It is not when all the rates are zero, a well that disallows crossflow should be SHUT. 

It is actually, when we have a zero rate target, a well that disallow crossflow should be SHUT. 

For example, a history well is under oil rate target control, while we have some non-zero water historical rate, although it is problematic, while it is not illegal. 

To extend, for prediction well, when we have a well that have zero rate constraint, it should also be SHUT if crossflow is disallowed. 

I think the logic applies to both injectors and producers.